pandas/core/indexes/base.py

```diff
@@ -2170,17 +2170,19 @@ def _assert_take_fillable(self, values, indices, allow_fill=True,
                               fill_value=None, na_value=np.nan):
         """ Internal method to handle NA filling of take """
         indices = _ensure_platform_int(indices)
-
         # only fill if we are passing a non-None fill_value
         if allow_fill and fill_value is not None:
             if (indices < -1).any():
                 msg = ('When allow_fill=True and fill_value is not None, '
                        'all indices must be >= -1')
                 raise ValueError(msg)
-            taken = values.take(indices)
             mask = indices == -1
-            if mask.any():
-                taken[mask] = na_value
+            if mask.all():
```

Comment:
@TomAugspurger did we move the empty check *into* ``take`` (the new helper)?

Reply:
Yeah, `algos.take` simplifies this. The only thing it doesn't do is the check on lines 2174-2178, where we raise a ValueError when

- indices has negative values
- allow_fill is true
- fill_value is specified

which seems like a strange set of conditions. Going to see what triggers it.